falcon/testing/test_case.py

```diff
@@ -18,36 +18,79 @@
 utilities for simulating and validating HTTP requests.
 """
 
-try:
-    import testtools as unittest
-except ImportError:  # pragma: nocover
-    import unittest
+import os
+import unittest
+import warnings
 
 import falcon
 import falcon.request
-
-# TODO hoist for backwards compat. Remove in falcon 4.
 from falcon.testing.client import Result  # NOQA
 from falcon.testing.client import TestClient
 
+base_case = os.environ.get('FALCON_BASE_TEST_CASE')
+
+if base_case and 'testtools.TestCase' in base_case:
+    try:
+        import testtools
+
+        BaseTestCase = testtools.TestCase
+    except ImportError:
+        BaseTestCase = unittest.TestCase
+elif base_case is None:
+    try:
+        import testtools
+
+        warnings.warn(
+            'Support for testtools is deprecated and will be removed in Falcon 5.0.',
```

Comment:
As requested before, please do not warn just because the package is importable!
Only warn when a `TestCase` is actually initialized.